source/install/trouble_mysql.rst

```diff
@@ -0,0 +1,268 @@
+    | mattermost         |
+    | mysql              |
+    | performance_schema |
+    | sys                |
+    +--------------------+
+    5 rows in set (0.03 sec)
+
+**No mattermost Database**
+
+If the ``mattermost`` database doesn't exist, create a database named
+``mattermost`` by opening MySQL as root and issuing the command: ::
+
+    create database mattermost;
+
+If you accidentally created a database with the wrong name, you can
+remove it by issuing the command: ::
+
+    drop database {misnamed};
+
+After creation of the database, attempt to restart the Mattermost server
+by navigating to the ``/opt/mattermost/bin`` directory and entering the
+command ``sudo -u mattermost ./platform``.
+
+**The mattermost Database Exists**
+
+If the ``mattermost`` database does exist, confirm that you have set up
+the database driver correctly in the
+``/opt/mattermost/config/config.json`` file. Open this file in a text
+editor, and review the value of ``"DataSource"``. It should be: ::
+
+    "mmuser:<mmuser-password>@tcp(<host-name-or-IP>:3306)/mattermost?charset-utf8mb4,utf8&readTimeout-30s&writeTimeout-30s"
+
+Replace `mmuser-password` with the password created during installation
+and `host-name-or-IP` with the appropriate value.
+You should also confirm that the ``DriverName`` element (found immediately
+above the ``DataSource`` element) is set to ``mysql``.
+
+If you correct an error, restart the Mattermost server by navigating to
+the ``/opt/mattermost/bin`` directory and entering the command
+``sudo -u mattermost ./platform``.
+
+The Database User
+~~~~~~~~~~~~~~~~~
+
+During installation, you create a MySQL database user from the *mysql*
+prompt by issuing the following command:::
+
+    create user 'mmuser'@'%' identified by '{mmuser-password}';`
+    
+The ``mmuser-password`` value is a placeholder for the password you chose.
+You may also have specified an IP address rather than the wild card
+``%``.
+
+.. note:: MySQL users are fully defined by their username and the host that they access MySQL from. These elements are separated by the ``@`` sign. The ``%`` character is a wild card indicating that the user can access MySQL from any IP address or host. If the user you created accesses MySQL from a specific host such as ``10.10.10.2``, please substitute that IP address for ``%``.
+
+If the user and host combination that you created does not exist, you
+will see an error such as: ::
+
+    [2017/09/20 17:06:18 EDT] [INFO] Pinging SQL master database
+    [2017/09/20 17:06:18 EDT] [EROR] Failed to ping DB retrying in 10 seconds 
+    err-Error 1045: Access denied for user 'mmuser'@'localhost' (using password: YES)
+
+**Checking that mmuser Exists**
+
+To check that this user exists log in to MySQL as the root user: ::
+
+    mysql -u root -p
+
+When prompted, enter the root password that you chose 
+when installing MySQL. From the ``mysql`` prompt enter the following command: ::
+
+    select User, Host from mysql.user;
```

Comment:
Maybe change this to `select User, Host from mysql.user where User = 'mmuser';` to more clearly show what the user should be seeing? They'll usually see the other accounts even if mmuser doesn't exist, so they might think that's close enough to what's shown in the example